File: 2134-MaximizeTheConfusionOfAnExam/2134-MaximizeTheConfusionOfAnExam.cpp
// Last updated: 6/11/2026, 11:24:46 AM
class Solution {
public:
    int maxConsecutiveAnswers(string s, int k) {
        int sz = s.size();
        int l = 0;
        int cnt = 0;
        int res = 0;
        for(int i=0; i<s.size(); i++){
            if(s[i] == 'T'){
                cnt++;
            }
            while(cnt>k){
                if(s[l] == 'T'){
                    cnt--;
                }
                l++;
            }
            res = max(res, i-l+1);
        }

        cnt = 0;

        l = 0;
        for(int i=0; i<s.size(); i++){
            if(s[i] == 'F'){
                cnt++;
            }
            while(cnt>k){
                if(s[l] == 'F'){
                    cnt--;
                }
                l++;
            }

            res = max(res, i-l+1);
        }

        return res;




    }
};

File: 2134-MaximizeTheConfusionOfAnExam/2134-MaximizeTheConfusionOfAnExam.py
# Last updated: 6/11/2026, 11:24:56 AM
class Solution:
    def maxConsecutiveAnswers(self, ans: str, k: int) -> int:
        
        # t to f
        def fun(chr):
            l = 0
            cnt = res = 0
            for r in range(len(ans)):
                if(ans[r] == chr):  
                    cnt += 1
                while(cnt>k):
                    if(ans[l] == chr):
                        cnt -= 1
                    l += 1
                
                res = max(res, r-l+1)
            return res
        return max(fun("T"), fun("F"))